### PR TITLE
http: moving functions and member variables out of http1 pool

### PR DESCRIPTION
**Pull request by alyssawilk**
As part of #3431 we need to move the logic from HTTP/1 and HTTP/2 connection pools to the active client,
so the new mixed connection pool can just create an active client of the right type.

This removes member variables from the HTTP/1.1 connection pool in preparation for that move.
a follow-up PR will do the same for HTTP/2, then the two classes will be removed in favor of the base http connection
pool being created with an instantiate_active_client_ closure which creates the right type of clients.

The alpn pool will eventually create an active client of the right type based on initial ALPN.

Risk Level: High (data plane refactor, albeit intended no-op)
Testing: existing tests pass
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/http/conn_pool_base.h [hunk 1]
```diff
@@ -69,6 +70,10 @@ class HttpConnPoolImplBase : public Envoy::ConnectionPool::ConnPoolImplBase,
                    Envoy::ConnectionPool::AttachContext& context) override;
 
   virtual CodecClientPtr createCodecClient(Upstream::Host::CreateConnectionData& data) PURE;
+
+protected:
+  Random::RandomGenerator& random_generator_;
+  Http::Protocol protocol_;
```
Comment (ggreenway) [inline source/common/http/conn_pool_base.h:76]: const?

File: source/common/http/conn_pool_base.h [hunk 2]
```diff
@@ -78,8 +83,13 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
                uint64_t concurrent_stream_limit)
       : Envoy::ConnectionPool::ActiveClient(parent, lifetime_stream_limit,
                                             concurrent_stream_limit) {
-    Upstream::Host::CreateConnectionData data = parent_.host()->createConnection(
-        parent_.dispatcher(), parent_.socketOptions(), parent_.transportSocketOptions());
+    Upstream::Host::CreateConnectionData data =
+        static_cast<Envoy::ConnectionPool::ConnPoolImplBase*>(&parent)->host()->createConnection(
```
Comment (ggreenway) [inline source/common/http/conn_pool_base.h:87]: What's going on with this cast? Isn't this an upcast, and thus would happen implicitly?

**Comment by alyssawilk** [comment]
@ggreenway with Snow out can you do first pass?

File: source/common/http/conn_pool_base.h [hunk 3]
```diff
@@ -78,8 +83,13 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
...
+            parent.dispatcher(), parent.socketOptions(), parent.transportSocketOptions());
+    initialize(data, parent);
+  }
+
+  void initialize(Upstream::Host::CreateConnectionData& data, HttpConnPoolImplBase& parent) {
```
Comment (ggreenway) [inline source/common/http/conn_pool_base.h:92]: Why is this now a separate function? I don't see it called anywhere else. Is this for a future PR?

**Review by ggreenway — CHANGES_REQUESTED** [review]

File: source/common/http/conn_pool_base.h (hunk 3 above)
Comment (alyssawilk) [reply source/common/http/conn_pool_base.h:92]: Yeah, there's going to be two constructors in a follow-up PR.
I can move it to the next but if you're OK I'm inclined to leave it.

File: source/common/http/conn_pool_base.h (hunk 1 above)
Comment (alyssawilk) [reply source/common/http/conn_pool_base.h:76]: it won't be for the ALPN subclass. :-)  OK if I leave it for now?

File: source/common/http/conn_pool_base.h (hunk 2 above)
Comment (alyssawilk) [reply source/common/http/conn_pool_base.h:87]: Annoyingly, 
HttpConnPoolImplBase has
Upstream::HostDescriptionConstSharedPtr host() const override { return host_; }

and the base class has
const Upstream::HostConstSharedPtr& host() const { return host_; }

so without the cast it snags the wrong type and complains there's no createConnection function.

**Review by alyssawilk — COMMENTED** [review]
Thanks for the review!

File: source/common/http/conn_pool_base.h (hunk 2 above)
Comment (ggreenway) [reply source/common/http/conn_pool_base.h:87]: Yuck. Can you add a comment to that effect?

Comment (alyssawilk) [reply source/common/http/conn_pool_base.h:89]: yeah - will add it to my clean up list too - may be able to do away with it with one of the later refactors!

**Review by ggreenway — APPROVED** [review]

**Review by ggreenway — APPROVED** [review]

**Pull request merged**